client/src/includes/initIconSprite.ts

```diff
@@ -0,0 +1,45 @@
+export const initIconSprite = (
+  spriteContainer: HTMLElement,
+  spriteURL: string,
+) => {
+  const revisionKey = 'wagtail:spriteRevision';
+  const dataKey = 'wagtail:spriteData';
+  const isLocalStorage: boolean =
+    'localStorage' in window && typeof window.localStorage !== 'undefined';
+  const insertIt = (data: string) => {
+    if (spriteContainer) {
+      // eslint-disable-next-line no-param-reassign
+      spriteContainer.innerHTML = data;
+    }
+  };
+
+  const insert = (data: string) => {
+    if (document.body) {
+      insertIt(data);
+    } else {
+      document.addEventListener('DOMContentLoaded', () => insertIt(data));
+    }
+  };
+
+  if (isLocalStorage && localStorage.getItem(revisionKey) === spriteURL) {
+    const data = localStorage.getItem(dataKey);
+    if (data) {
+      insert(data);
+      return;
+    }
+  }
```

Comment:
A refinement here, let's pull out the `insertIt` behaviour into two parts, one that works out if the DOM has loaded and then the code to actually update the DOM becomes much simpler.

```suggestion
  const insert = (data: string | null) => {
    if (!spriteContainer || !data) return;

    domReady().then(() => {
      // eslint-disable-next-line no-param-reassign
      spriteContainer.innerHTML = data;
    });
  };
```

The function `domReady` would be this...
```TypeScript
const domReady = () =>
  new Promise<void>((resolve) => {
    if (document.readyState === 'loading') {
      document.addEventListener('DOMContentLoaded', () => resolve(), {
        once: true,
        passive: true,
      });
    } else {
      resolve();
    }
  });
```